### ISSUE TITLE
jsdom test fails

### ISSUE THREAD

**Issue by elcontraption** [Tribex/prerenderer#7]
Running `npm run test:jsdom` fails with an undefined `window`. Does a server need to be running the app already or should the jsdom renderer take care of that?

```
> @prerenderer/prerenderer@0.6.6 test:jsdom /[path]/prerenderer
> jest ./test/**/*.jsdom.test.js --silent --runInBand


 RUNS  test/basic/basic.jsdom.test.js
/[path]/prerenderer/renderers/renderer-jsdom/es6/renderer.js:99
            window.addEventListener('error', function (event) {
                   ^

TypeError: Cannot read property 'addEventListener' of undefined
    at Object.created (/[path]/prerenderer/renderers/renderer-jsdom/es6/renderer.js:99:20)
    at reportInitError (/[path]/prerenderer/node_modules/jsdom/lib/old-api.js:472:12)
    at resourceLoader.download (/[path]/prerenderer/node_modules/jsdom/lib/old-api.js:291:9)
    at Request.request [as _callback] (/[path]/prerenderer/node_modules/jsdom/lib/jsdom/browser/resource-loader.js:208:7)
    at self.callback (/[path]/prerenderer/node_modules/request/request.js:186:22)
    at Request.emit (events.js:160:13)
    at Request.Object.<anonymous>.Request.onRequestError (/[path]/prerenderer/node_modules/request/request.js:878:8)
    at ClientRequest.emit (events.js:160:13)
    at Socket.socketErrorListener (_http_client.js:389:9)
    at Socket.emit (events.js:160:13)
    at emitErrorNT (internal/streams/destroy.js:64:8)
    at process._tickCallback (internal/process/next_tick.js:152:19)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @prerenderer/prerenderer@0.6.6 test:jsdom: `jest ./test/**/*.jsdom.test.js --silent --runInBand`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @prerenderer/prerenderer@0.6.6 test:jsdom script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /[path]/.npm/_logs/2018-02-27T16_24_49_190Z-debug.log
```

**Comment by elcontraption** [comment]
Update: appears to work when rendering a smaller number of routes:

```js
// /test/basic/basic.jsdom.test.js

// breaks:
// const routes = new Array(1000).fill().map(i => generateRandomRoute())

// works:
const routes = new Array(100).fill().map(i => generateRandomRoute())
```

**Comment by Tribex** [comment]
@elcontraption Odd, looks like a race condition. Good catch! Oddly enough, I can't seem to reproduce. O-o

**Comment by Tribex** [comment]
Should be fixed in 0b1ac12390a80a26d3599a603df3eab2474a0ea6 (or rather, you'll get an `ECONNRESET` error now).
It wasn't handling JSDOM initialization errors properly. It looks like either JSDOM or the server was getting swamped. This can be avoided in production by using the `maxConcurrentRoutes` option.
